microcosm_pubsub/dispatcher.py

```diff
@@ -110,6 +111,9 @@ def handle_message(self, message, bound_handlers) -> MessageHandlingResult:
                 try:
                     self.validate_message(message)
                     handler = self.find_handler(message, bound_handlers)
+                    self.opaque["X-Request-Handler"] = handler.__name__
```

Comment:
Assigning to opaque directly is probably ok.

We want this info to get logged by web services, so it needs to be passed as a header from marquez in a form that won't get scrubbed by microcosm-flask. That is why we use the "X-Request-" prefix

See:
- https://github.com/globality-corp/marquez/blob/master/marquez/http_client.py#L69
- https://github.com/globality-corp/microcosm-flask/blob/develop/microcosm_flask/context.py#L17